### PR TITLE
Question regarding Series.argsort documentation #24816

### LINKED ISSUE #24816: Question regarding Series.argsort documentation
`Series.argsort` has three arguments, `axis`, `kind` and `order`, but only `kind` is actually used.  Should the other two be removed from the documentation? (I'm guessing they were once used and are left in the code for compatibility reasons?)

https://pandas.pydata.org/pandas-docs/stable/generated/pandas.Series.argsort.html

### PR DESCRIPTION
**Pull request by benjaminr**
- [ ] closes #24816 
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Review by TomAugspurger — APPROVED** [review]

File: pandas/core/series.py [hunk 1]
```diff
@@ -2993,10 +2993,13 @@ def argsort(self, axis=0, kind='quicksort', order=None):
         Parameters
         ----------
         axis : int (can only be zero)
```
Comment (jschendel) [inline pandas/core/series.py]: The "(can only be zero)" can be removed, as it doesn't appear that this is checked:
```python
In [2]: s = pd.Series([3, 1, 4, 1, 5])

In [3]: s.argsort(axis=10)
Out[3]:
0    1
1    3
2    0
3    2
4    4
dtype: int64
```

File: pandas/core/series.py [hunk 2]
```diff
@@ -2993,10 +2993,13 @@ def argsort(self, axis=0, kind='quicksort', order=None):
...
+            Has no effect but is accepted for compatibility with numpy.
         kind : {'mergesort', 'quicksort', 'heapsort'}, default 'quicksort'
             Choice of sorting algorithm. See np.sort for more
             information. 'mergesort' is the only stable algorithm
-        order : ignored
+        order : None
+            Must be None. Has no effect but is accepted for compatibility
```
Comment (jschendel) [inline pandas/core/series.py]: The "Must be None" can be removed, as it doesn't appear that this is checked:
```python
In [4]: s.argsort(order='foo')
Out[4]:
0    1
1    3
2    0
3    2
4    4
dtype: int64
```

**Review by jschendel — COMMENTED** [review]
Thanks! Looks good overall. A few small comments.

**Review by jschendel — APPROVED** [review]

**Comment by jschendel** [comment]
Thanks @benjaminr!

**Pull request merged**